plugins/github-pull-requests-board/src/hooks/useUserRepositoriesAndTeam.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import { catalogApiRef, useEntity } from '@backstage/plugin-catalog-react';
+import { useCallback, useEffect, useState } from 'react';
+import {
+  getProjectNameFromEntity,
+  getGithubOrganizationFromEntity,
+  getUserNameFromEntity,
+} from '../utils/functions';
+
+export function useUserRepositoriesAndTeam() {
+  const { entity: teamEntity } = useEntity();
+  const catalogApi = useApi(catalogApiRef);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [teamData, setTeamData] = useState<{
+    repositories: string[];
+    teamMembers: string[];
+  }>({
+    repositories: [],
+    teamMembers: [],
+  });
+
+  const getTeamData = useCallback(async () => {
```

Comment:
In general, `useAsync` (see elsewhere in the repo for tons of examples) is an excellent helper for doing any async loading. It'll keep track of loading, error, and value automatically for you, so you can remove all of the useState things as well as the useEffect